storage/src/vespa/storage/storageserver/rpc/caching_rpc_target_resolver.cpp

```diff
@@ -1,82 +1,107 @@
 // Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 #include "caching_rpc_target_resolver.h"
 #include "shared_rpc_resources.h"
-#include <vespa/fnet/frt/supervisor.h>
 #include <vespa/fnet/frt/target.h>
-#include <vespa/slobrok/sbmirror.h>
+#include <vespa/slobrok/imirrorapi.h>
 #include <vespa/storageapi/messageapi/storagemessage.h>
 #include <vespa/vespalib/stllike/asciistream.h>
 #include <vespa/vespalib/stllike/hash_map.hpp>
+#include <cassert>
 
 #include <vespa/log/log.h>
 LOG_SETUP(".storage.caching_rpc_target_resolver");
 
 namespace storage::rpc {
 
-CachingRpcTargetResolver::CachingRpcTargetResolver(SharedRpcResources& rpc_resources)
-    : _rpc_resources(rpc_resources)
+CachingRpcTargetResolver::CachingRpcTargetResolver(const slobrok::api::IMirrorAPI& slobrok_mirror,
+                                                   const RpcTargetFactory& target_factory)
+    : _slobrok_mirror(slobrok_mirror),
+      _target_factory(target_factory),
+      _targets_rwmutex()
 {
 }
 
 CachingRpcTargetResolver::~CachingRpcTargetResolver() = default;
 
-namespace {
-
-vespalib::string address_to_slobrok_id(const api::StorageMessageAddress& address) {
+vespalib::string
+CachingRpcTargetResolver::address_to_slobrok_id(const api::StorageMessageAddress& address) {
     vespalib::asciistream as;
     as << "storage/cluster." << address.getCluster()
        << '/' << ((address.getNodeType() == lib::NodeType::STORAGE) ? "storage" : "distributor")
        << '/' << address.getIndex();
     return as.str();
 }
 
+std::shared_ptr<RpcTarget>
+CachingRpcTargetResolver::lookup_target(const vespalib::string& slobrok_id, uint32_t curr_slobrok_gen) {
+    std::shared_lock lock(_targets_rwmutex);
+    auto itr = _targets.find(slobrok_id);
+    if ((itr != _targets.end())
+        && itr->second->_target->is_valid()
+        && (itr->second->_slobrok_gen == curr_slobrok_gen)) {
+        return itr->second;
+    }
+    return {};
+}
+
+std::shared_ptr<RpcTarget>
+CachingRpcTargetResolver::consider_update_target(const vespalib::string& slobrok_id,
+                                                 const vespalib::string& connection_spec,
+                                                 uint32_t curr_slobrok_gen,
+                                                 const UniqueLock& targets_lock) {
+    (void) targets_lock;
```

Comment:
Could also use `[[maybe_unused]]` directly on the parameter itself

Reply:
Fixed